File: codeburn/mac/Sources/CodeBurnMenubar/SessionCountLabel.swift
/// Shared session-count phrasing. Keep in lockstep with `src/session-count-label.ts`.
enum SessionCountLabel {
    static let help = "Older session logs may be unavailable."
    /// Combined-scope counts are a per-device numeric sum with no shared identity.
    /// Do not show that sum as unique or as a lower bound.
    static let combinedHelp = "Session identities are unavailable across devices."
    static let combinedText = "Session count unavailable"

    static func isExact(_ basis: String?) -> Bool {
        basis == "identity"
    }

    static func text(sessions: Int, basis: String?) -> String {
        if !isExact(basis) {
            if sessions <= 0 { return "Session count unavailable" }
            return sessions == 1 ? "At least 1 session" : "At least \(sessions) sessions"
        }
        return sessions == 1 ? "1 session" : "\(sessions) sessions"
    }

    static func compact(sessions: Int, basis: String?) -> String {
        if !isExact(basis) {
            if sessions <= 0 { return "Unavailable" }
            return "≥\(sessions) sess"
        }
        return "\(sessions) sess"
    }

    static func averageText(_ value: Double?, basis: String?, format: (Double) -> String) -> String {
        guard isExact(basis), let value, value.isFinite else { return "—" }
        return format(value)
    }

    static func helpText(combined: Bool, basis: String?) -> String {
        if combined { return combinedHelp }
        return isExact(basis) ? "" : help
    }
}

File: codeburn/src/session-count-label.ts
/// Shared session-count phrasing for CLI, MCP, and tests.
/// Exact unique counts come only from surviving source identities with no
/// unknown cache contribution. Everything else is a lower bound.
export type SessionCountBasis = 'identity' | 'partial'

export const SESSION_COUNT_HELP = 'Older session logs may be unavailable.'

export function sessionCountIsExact(basis: SessionCountBasis | undefined): boolean {
  return basis === 'identity'
}

export function formatSessionCount(
  sessions: number,
  basis: SessionCountBasis | undefined,
): string {
  if (!sessionCountIsExact(basis)) {
    if (sessions <= 0) return 'Session count unavailable'
    return sessions === 1 ? 'At least 1 session' : `At least ${sessions.toLocaleString('en-US')} sessions`
  }
  if (sessions === 1) return '1 session'
  return `${sessions.toLocaleString('en-US')} sessions`
}
